redis/asyncio/connection.py

```diff
@@ -222,10 +222,15 @@ def is_connected(self):
         return self._reader is not None and self._writer is not None
 
     def register_connect_callback(self, callback):
-        self._connect_callbacks.append(weakref.WeakMethod(callback))
+        wm = weakref.WeakMethod(callback)
+        if wm not in self._connect_callbacks:
+            self._connect_callbacks.append(wm)
 
-    def clear_connect_callbacks(self):
-        self._connect_callbacks = []
+    def deregister_connect_callback(self, callback):
```

Comment:
Given the single use, I'd argue that this isn't a breaking change - but as it's meant to be internal to the library and not reused, let's fix that here. WDYT about an underscore prefix? Yes, I could argue that register_connect_callback might have the same need :(

Reply:
fair point

Reply:
Done, also for non-async to keep consistency.